de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/session/internal/SharedReferencePointMapper.java

```diff
@@ -0,0 +1,543 @@
+
+        LOG.debug("removed referencePoint " + referencePoint + " with ID " + id);
+    }
+
+    /**
+     * Adds the given resources to a <b>partially</b> shared reference point.
+     * 
+     * @param referencePoint
+     *            a referencePoint that was added as a partially shared
+     *            referencePoint
+     * @param resources
+     *            the resources to add
+     */
+    /*
+     * TODO needs proper sync. in the SarosSession class
+     * 
+     * @throws IllegalStateException if the project is completely or not shared
+     * at all
+     */
+    public synchronized void addResources(IReferencePoint referencePoint,
+        Collection<? extends IResource> resources) {
+
+        if (referencePointToIDMapping.get(referencePoint) == null) {
+            LOG.warn("could not add resources to referencePoint "
+                + referencePoint + " because it is not shared");
+            // throw new IllegalStateException(
+            // "could not add resources to project " + project
+            // + " because it is not shared");
+            return;
+        }
+
+        if (completelySharedReferencePoints.contains(referencePoint)) {
+            LOG.warn("cannot add resources to completely shared referencePoint: "
+                + referencePoint);
+            // throw new IllegalStateException(
+            // "cannot add resources to completely shared project: " + project);
+            return;
+        }
+
+        Set<IResource> partiallySharedResources = partiallySharedResourceMapping
+            .get(referencePoint);
+
+        if (partiallySharedResources.isEmpty()) {
+            partiallySharedResources = new HashSet<IResource>(Math.max(1024,
+                (resources.size() * 3) / 2));
+
+            partiallySharedResourceMapping.put(referencePoint,
+                partiallySharedResources);
+        }
+
+        partiallySharedResources.addAll(resources);
+    }
+
+    /**
+     * Removes the given resources from a <b>partially</b> shared reference
+     * point.
+     * 
+     * @param referencePoint
+     *            a reference point that was added as a partially shared
+     *            reference point
+     * @param resources
+     *            the resources to remove
+     */
+    /*
+     * TODO needs proper sync. in the SarosSession class
+     * 
+     * @throws IllegalStateException if the referencePoint is completely or not
+     * shared at all
+     */
+    public synchronized void removeResources(IReferencePoint referencePoint,
+        Collection<? extends IResource> resources) {
+
+        if (referencePointToIDMapping.get(referencePoint) == null) {
+            LOG.warn("could not remove resources from referencePoint "
+                + referencePoint + " because it is not shared");
+            // throw new IllegalStateException(
+            // "could not remove resources from project " + project
+            // + " because it is not shared");
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: This block of commented-out lines of code should be removed. ([squid:CommentedOutCodeLine](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ACommentedOutCodeLine))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=257&issues=AWbeHvN4SgN4uCclRsGz&open=AWbeHvN4SgN4uCclRsGz)

[ISSUE_KEY]: AWbeHvN4SgN4uCclRsGz